internal/commands/cnab.go

```diff
@@ -35,41 +39,94 @@ type bindMount struct {
 
 const defaultSocketPath string = "/var/run/docker.sock"
 
-func prepareCredentialSet(contextName string, contextStore store.Store, b *bundle.Bundle, namedCredentialsets []string) (map[string]string, error) {
+type credentialSetOpt func(b *bundle.Bundle, creds map[string]string) error
+
+func addNamedCredentialSets(namedCredentialsets []string) credentialSetOpt {
+	return func(_ *bundle.Bundle, creds map[string]string) error {
+		for _, file := range namedCredentialsets {
+			if _, err := os.Stat(file); err != nil {
+				file = filepath.Join(duffleHome().Credentials(), file+".yaml")
+			}
+			c, err := credentials.Load(file)
+			if err != nil {
+				return err
+			}
+			values, err := c.Resolve()
+			if err != nil {
+				return err
+			}
+			for k, v := range values {
+				if _, ok := creds[k]; ok {
+					return fmt.Errorf("ambiguous credential resolution: %q is present in multiple credential sets", k)
```

Comment:
Shouldn't we move this check to duffle? It feels like everybody will need it.

Reply:
It couldn't just be the check, it'd have to be the whole of this merge-named-creds-into-a-set functionality I think.

Reply:
Yep indeed, that's what I meant 😅

Reply:
On it now.

Reply:
https://github.com/deislabs/duffle/pull/682 adds such a helper and this PR now uses it.